docs/design.md

```diff
@@ -220,6 +220,24 @@ and during development, as well as any desired safety margin on the
 manufacturer's specified number of erase cycles. In general, using a ratio that
 allows hundreds to thousands of field upgrades in production is recommended.
 
+## [swap without using scratch](#image-swap-no-scratch)
+
+This algorithm is alternatively to the swap using scratch algorithm. It uses
+additional sector in the primary slot to make swap possible. Algorithm works
+as fallow:
+
+  1.	Move all swap of the primary image up by one swap.  
+    Beginning form N=0:
+  2.	copy Nth sector from secondary slot to Nth sector of primary slot
+  3.	copy (N+1)th sector from primary slot to the Nth secondary slot
+  4.	repeat steps 2. and 3. unless entire images are copied
+
+This algorithm is designed in such way that primary slot should be exact one
```

Comment:
The algorithm allows both slots to be have the same size as well as allowing a primary slot with secondary+1 sectors, so this sentence is not entirely correct.

Reply:
^^ right, have to rephrase -my intention was emphasize what is most efective